zathura/shortcuts.c

```diff
@@ -76,6 +76,61 @@ static bool link_shortcuts(zathura_t* zathura, girara_callback_inputbar_activate
   return false;
 }
 
+/* Helper function to redo the search and stay in place */
+bool redo_search(zathura_t* zathura){
+    if (zathura->document == NULL || strlen(zathura->global.search_string) == 0) {
```

Comment:
First, this function should be static. Second, it is never called with document == NULL - otherwise the call site is buggy. The string length check could also be done at the call site where it is checked if search_string != NULL.